packages/zoe/src/contracts/helpers/userFlow.js

```diff
@@ -9,11 +11,44 @@ const hasKinds = (kinds, newPayoutRules) =>
 const hasBrands = (brands, newPayoutRules) =>
   brands.every((brand, i) => brand === newPayoutRules[i].amount.brand);
 
-export const makeHelpers = (zoe, issuers) => {
+const getKeysIf = obj =>
+  obj !== undefined ? harden(Object.getOwnPropertyNames(obj)) : harden([]);
+
+export const makeHelpers = (zoe, roles, issuers) => {
   const amountMathArray = zoe.getAmountMathForIssuers(issuers);
   const brands = zoe.getBrandsForIssuers(issuers);
   const zoeService = zoe.getZoeService();
+  const rejectIf = (
+    inviteHandle,
+    actual,
+    expected,
+    msg = defaultAcceptanceMsg,
+  ) => {
+    if (expected !== undefined) {
+      if (!sameStructure(getKeysIf(actual), expected)) {
+        zoe.complete(harden([inviteHandle]));
+        throw new Error(msg);
+      }
+    }
+  };
   const helpers = harden({
+    assertRoleNames: (actual, expected) => {
```

Comment:
normally `actual` and `expected` should have the same "structure". In this case, it looks like `actual` is a expected to be an already-sorted array? That seems surprising, so perhaps your should rename to "assertKeys" or some such?

Reply:
Thanks. I changed it to only take `expected`